esp/logging/loggingagentbase.hpp

```diff
@@ -112,8 +113,43 @@ interface IEspLogAgent : extends IInterface
 {
     virtual bool init(const char * name, const char * type, IPropertyTree * cfg, const char * process) = 0;
```

Comment:
we should add a "providesTransactionID" method. If a negative, then ESP knows not to query logagent for transactionid